src/bootstrap/src/core/config/config.rs

```diff
@@ -694,7 +688,9 @@ impl Config {
 
         let initial_rustc = build_rustc.unwrap_or_else(|| {
             download_beta_toolchain(&dwn_ctx, &out);
-            out.join(host_target).join("stage0").join("bin").join(exe("rustc", host_target))
+            let target = if cfg!(test) { get_host_target() } else { host_target };
```

Comment:
I don't like this change, tbh. I assume it's because of `configure_with_args`? Could we change that function to not override the `--build` flag instead? We already use the actual host as the host target in snapshot tests.

In general, I don't want to add a lot of custom code *just* to make `cargo test` work.